packages/block-editor/src/components/iframe/index.js

```diff
@@ -31,6 +31,30 @@ import { useWritingFlow } from '../writing-flow';
 import { useCompatibilityStyles } from './use-compatibility-styles';
 import { store as blockEditorStore } from '../../store';
 
+function bubbleEvent( event, Constructor, frame ) {
+	const init = {};
+
+	for ( const key in event ) {
+		init[ key ] = event[ key ];
+	}
+
+	if ( event instanceof frame.ownerDocument.defaultView.MouseEvent ) {
+		const rect = frame.getBoundingClientRect();
+		init.clientX += rect.left;
+		init.clientY += rect.top;
+	}
```

Comment:
I'm just looking into the issue over in https://github.com/WordPress/gutenberg/issues/55074:

Why is this guarded against a check for `event instanceof frame.ownerDocument.defaultView.MouseEvent`? I notice if I change this to `if ( frame ) {` I can get the drag position of the block drag chip (i.e. dragging from the inserter over into the editor canvas) working properly again. So it seems this is returning `false` for some situations where we still want the offset to be applied, so that drag positioning is correct. In this example, I think it might mean drag events that originated from outside the iframe?

Reply:
I don't really know, this code was there long before my PR. I'm guessing we should do this for all events that have "clientX" and "clientY" (we may want to exclude keyboard events for instance...)

Reply:
Thanks for confirming, I'll have a play around 🙂